src/core/public/chrome/ui/header/nav_link.tsx

```diff
@@ -134,16 +131,10 @@ export function createRecentNavLink(
 
   return {
     href,
-    label,
+    baseUrl: href,
```

Comment:
Is it a correct baseUrl value?

Reply:
I think `baseUrl` can be considered as a placeholder rather than a real value. The reason is that Recent ChromeNavLink is only consumed by function `createEuiListItem`, where `baseUrl` is not used.

Reply:
OK. Can we add some annotations about that? It may be confused.

Reply:
Added

Reply:
`baseUrl` is supposed to be `the base route used to open the root of an application`, in this case, we are trying to generate `ChromeNavLink` from recently accessed url and ofc we don't know what's the `root of an application`.

Instead of simply making an assumption:
> baseUrl is not used in this case, so we don't care if the value is correct or not

Here is what I'd suggest:

```typescript
// src/core/public/chrome/ui/header/nav_link.tsx
export type RecentNavLink = Omit<ChromeNavLink, 'baseUrl'>;

export function createRecentChromeNavLink(/*....*/): RecentNavLink {
  //...
}

// src/core/public/chrome/ui/header/collapsible_nav.tsx
const navLinks = useObservable(observables.navLinks$, []).filter((link) => !link.hidden);
const allNavLinks: Array<ChromeNavLink | RecentNavLink> = [...navLinks];
const recentlyAccessed = useObservable(observables.recentlyAccessed$, []);
if (recentlyAccessed.length) {
  allNavLinks.push(
    ...recentlyAccessed.map((link) => createRecentChromeNavLink(link, navLinks, basePath))
  );
} else {
  allNavLinks.push(emptyRecentlyVisited);
}
```

Reply:
Do we need another type for const `emptyRecentlyVisited`?

Reply:
Since `baseUrl` prop in `navLinks` is also not consumed, how about declaring a type
`export type CollapsibleNavLink = Omit<ChromeNavLink, 'baseUrl'>;`
And then get `navLinks` by:
`const navLinks = useObservable(observables.navLinks$, []).filter((link) => !link.hidden).map(chromeNavLinkToCollapsibleNavLink;`

Reply:
Not necessary, I would suggest you keep the existing code as it is if you can.

Reply:
Ok, I define a new type `ChromeOrRecentNavLink` in my latest commit to refactor the code. Please take a look
```
// src/core/public/chrome/ui/header/nav_link.tsx
export type ChromeOrRecentNavLink = ChromeNavLink | RecentNavLink;

// src/core/public/chrome/ui/header/collapsible_nav.tsx
function getAllCategories(allCategorizedLinks: Record<string, ChromeOrRecentNavLink[]>) {
```